### DISCUSSION TITLE
The wiki about windowrule workspace field is not clear

### DISCUSSION THREAD

**Discussion by forestchen1224** [hyprwm/Hyprland#3974] [Q&A] [answered]
I have a special workspace that is named 'test',and I want it's opactity to be 0,5.If I use windowrule to do that.what should be the **WINDOW** field?

The supported fields of windowrule mentioned in the  [wiki](https://wiki.hyprland.org/Configuring/Window-Rules/) are:
```class - class regex    
title - title regex  
xwayland - 0/1  
floating - 0/1  
fullscreen - 0/1  
pinned - 0/1  
workspace - id or name: and name 
 ```

When using an id(-96, for example),which is got from `hyprctl clients`,the rule  
```
windowrulev2 = opacity 0.5 override,workspace:-96
```
 works.
But since the id of special workspace often changes,I want to use the name `test`,then,
neither
```
windowrulev2 = opacity 0.5 override,workspace:test
```
,
```
windowrulev2 = opacity 0.5 override,workspace:(special:test)
```
nor 
```
windowrulev2 = opacity 0.5 override,workspace:special
```
works.

What is the problem or what have I missed?

**Comment by vaxerski** [comment]
have you tried `workspace:special:test`?

**Reply by forestchen1224** [reply]
Yes,I tried,still not working.

**Reply by forestchen1224** [reply]
By the way,I use archlinux, and the hyprland version I get from pacman is `0.30.0-1`

**Reply by vaxerski** [reply]
that is very outdated.

**Reply by forestchen1224** [reply]
I tried the latest release v0.32.3,the result is the same.

**Comment by forestchen1224** [comment]
Is there any update?

**Comment by forestchen1224** [comment, marked as answer]
The rule should be 
`windowrulev2 = opacity 0.5 override,workspace:name:special:test`
source code [here](https://github.com/hyprwm/Hyprland/blob/7e033e48ace5406a9bc442f7d403f9ce3af193f3/src/config/ConfigManager.cpp#L2101)
